src/easylink/pipeline_graph.py

```diff
@@ -60,11 +33,40 @@ def implementation_nodes(self) -> List[str]:
     @property
     def implementations(self) -> List[Implementation]:
         """Convenience property to get all implementations in the graph."""
-        return [self.get_attr(node, "implementation") for node in self.implementation_nodes]
+        return [self.nodes[node]["implementation"] for node in self.implementation_nodes]
 
-    def get_attr(self, node: str, attr: str) -> Any:
-        """Convenience method to get a particular attribute from a node"""
-        return self.nodes[node][attr]
+    def update_slot_filepaths(self, config: Config) -> None:
```

Comment:
Use the implementation metadata and the input data config to fill in the actual list of files that move from one implementation to the next.